File: DSA/Binary Search/Assignment 11/Question01.cpp
#include <iostream>
using namespace std;

int sqRoot(int x) {
    int s = 0;
    int e = x;

    long long int ans = -1;
    while (s <= e) {
        long long int mid = s + (e - s) / 2;
        long long int midMul = mid * mid;

        if (midMul == x) {
            return mid;
        } else if (midMul < x) {
            ans = mid;
            s = mid + 1;
        } else {
            e = mid - 1;
        }
    }

    return ans;
}

int main() {

    cout << sqRoot(4) << endl;
    cout << sqRoot(9) << endl;

    return 0;
}

File: Surprise Test/Test_02/Question01.js
function sqRoot(x) {
    let s = 0;
    let e = x;

    let ans = -1;
    while (s <= e) {
        let mid = Math.floor((e + s) / 2);
        let midMul = mid * mid;

        if (midMul === x) {
            return mid;
        } else if (midMul < x) {
            ans = mid;
            s = mid + 1;
        } else {
            e = mid - 1;
        }
    }

    return ans;
}

console.log(sqRoot(9))
console.log(sqRoot(12))
console.log(sqRoot(8))
